rs/execution_environment/src/execution_environment.rs

```diff
@@ -986,6 +986,41 @@ impl ExecutionEnvironment {
                 Ok(args) => self.deposit_cycles(args.get_canister_id(), &mut msg, &mut state),
             },
 
+            Ok(Ic00Method::FlexibleHttpRequest) => {
+                match state.metadata.own_subnet_features.http_requests {
+                    true => match &msg {
+                        CanisterCall::Request(_) => {
+                            match FlexibleCanisterHttpRequestArgs::decode(payload) {
+                                Err(err) => ExecuteSubnetMessageResult::Finished {
+                                    response: Err(err),
+                                    refund: msg.take_cycles(),
+                                },
+                                Ok(_) => ExecuteSubnetMessageResult::Finished {
+                                    response: Err(UserError::new(
```

Comment:
We want to provide a structured error type for this endpoint. Does it make sense to do so immediately, even if we return an error to the user? Do we want to make any guarantees about the types of reject codes that are possible or not possible?

Reply:
I guess it's consistent with the other calls, even if our overall error handling story is a bit of a mess. And I wouldn't be surprised if we end up using reject responses for some errors anyways even if the method itself is correct (like being unable to parse the Candid, for example). So fine with me.